addons/fire/config.cpp

```diff
@@ -1,5 +1,12 @@
 #include "script_component.hpp"
 
+#if __has_include("\z\ace\addons\nomedical\script_component.hpp")
+#define PATCH_SKIP "No Medical"
+#endif
+
+#ifdef PATCH_SKIP
+ACE_PATCH_NOT_LOADED(ADDON,PATCH_SKIP)
+#else
```

Comment:
```suggestion
#if __has_include("\z\ace\addons\nomedical\script_component.hpp")
ACE_PATCH_NOT_LOADED(ADDON,PATCH_SKIP)
#else

```

Why not just this? :thinking:

Reply:
This way can define `PATCH_SKIP` with several conditions. E.g.:
```
#if __has_include("\z\ace\addons\nomedical\script_component.hpp")
#define PATCH_SKIP "No Medical"
#endif
#if __has_include("\some\other_addon\config.bin")
#define PATCH_SKIP "Some Other Addon"
#endif
```
and then you use `PATCH_SKIP` just once to disable component with `ACE_PATCH_NOT_LOADED`. In this PR this is mostly a good template for other `__has_include` components. I liked @PabstMirror idea and implemeted it here too.
See also https://github.com/acemod/ACE3/blob/fdbcc62c74f1c51cf3ff36b784789a92fada4702/addons/compat_ws/config.cpp#L3-L9

Also you have `PATCH_SKIP` undefined in suggestion.

Reply:
Ok your example adds multiple conditions, but you can also do that like this:
```cpp
#if __has_include("\z\ace\addons\nomedical\script_component.hpp")
#if __has_include("\some\other_addon\config.bin")
ACE_PATCH_NOT_LOADED(ADDON,"Some Other Addon")
#else
ACE_PATCH_NOT_LOADED(ADDON,"No Medical")
#else
// default config
#endif
#endif
```
Although that is uglier and I wouldn't want to do it.

Not sure, we just need the opposite of `skipWhenMissingDependencies`!

_Fixed suggestion._

Reply:
I am not fully sure about `#if`, but I seem to remember that preprocessor "commands" like that are not able to be nested(at least `#ifdef` is not able to be nested)

Reply:
Ah, and they also don't support logical operators I believe. Yeah, we need a `skipWhenLoaded`.

Reply:
```
#if __has_include("\rhsafrf\addons\rhs_main\loadorder\config.bin")
#else
    #define PATCH_SKIP "RHS AFRF"
#endif

#if __has_include("\z\ace\addons\explosives\script_component.hpp")
#else
    #ifdef PATCH_SKIP
    #else
        #define PATCH_SKIP "ACE Explosives"
    #endif
#endif
```
works on game, mikro and hemtt

Reply:
What about [`#ifndef`](https://community.bistudio.com/wiki/PreProcessor_Commands#.23ifndef), is it working everywhere ? Cuz if it is, I don't see why do ifdef-else.

Reply:
>  but I seem to remember that preprocessor "commands" like that are not able to be nested(at least #ifdef is not able to be nested)

This should be fixed at least for a patch or two.

Reply:
#9294 seems to work fine with nesting.